### ISSUE TITLE
Whitespace in variable assignment not detected

### ISSUE THREAD

**Issue by inianv** [koalaman/shellcheck#874]
Expected rule-id: [SC1068]( https://github.com/koalaman/shellcheck/wiki/SC1068)
checked on the http://www.shellcheck.net/ web version

#### Here's a snippet or screenshot that shows the problem:

```
#!/usr/bin/bash
filename="$1"
count=0
file=0
br=$(wc -l $filename | awk '{print $1}')
let br = $br/10
while read -r line
do
let count = count + 1
    name="$line"
echo $name >> file$file.csv
if [ $count = $br ];then
let count=0
let file+=1
fi
done < "$filename"
```

The line 

let count = count + 1

should have thrown the error.

**Comment by kurahaupo** [comment]
On Wed, 12 Apr 2017, inianv wrote:
> The line
>     let count = count + 1
> should have thrown the error.

It's not the fact that it's an assignment with spaces that matters.

Rather, each word presented to "let" is taken a separate expression.

I suppose one could check that an expression does not start or end with "=",
but something like `let a=b +c++` is quite valid -- though unlikely to be
what was intended.

Conversely `let "count = count + 1"` is valid and does what you'd expect.

Perhaps we should just say "let is deprecated, use ((...)) instead"?
